From sshtylyov@ru.mvista.com  Sat Jul 18 15:46:20 2009
From: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Date: Fri, 17 Jul 2009 17:30:03 +0300
Subject: USB: musb_gadget_ep0: fix typo in service_zero_data_request()
To: gregkh@suse.de, david-b@pacbell.net
Cc: linux-usb@vger.kernel.org, felipe.balbi@nokia.com
Message-ID: <200907171830.03495.sshtylyov@ru.mvista.com>
Content-Disposition: inline


This function uses wrong bit mask to prevent clearing RXCSR status
bits when halting an endpoint -- which results in clearing SentStall
and RxPktRdy bits (that the code actually tries to avoid); must be
a result of cut-and-paste...

Signed-off-by: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/musb/musb_gadget_ep0.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/musb/musb_gadget_ep0.c
+++ b/drivers/usb/musb/musb_gadget_ep0.c
@@ -407,7 +407,7 @@ stall:
 					csr |= MUSB_RXCSR_P_SENDSTALL
 						| MUSB_RXCSR_FLUSHFIFO
 						| MUSB_RXCSR_CLRDATATOG
-						| MUSB_TXCSR_P_WZC_BITS;
+						| MUSB_RXCSR_P_WZC_BITS;
 					musb_writew(regs, MUSB_RXCSR,
 							csr);
 				}
